src/Text/Pandoc/Writers/ConTeXt.hs

```diff
@@ -93,6 +93,9 @@ pandocToConTeXt options (Pandoc meta blocks) = do
                         Just (('a':d:ds) :: String)
                           | all isDigit (d:ds) -> resetField "papersize"
                                                      (('A':d:ds) :: String)
+                        _                     -> id)
+                $ (case getField "pdfa" metadata of
+                        Just ("true" :: String) -> resetField "pdfa" ("1b:2005" :: String)
                         _                     -> id) metadata
```

Comment:
Pandoc metadata can be set using a confusing number of ways... this only checks for the string "true", but we also have actual booleans... but this should do the trick:

```haskell
import Data.Char (ord, isDigit, toLower)
-- rest of the file...

                $ (case toLower <$> (lookupMetaString "pdfa" meta) of
                        "true" -> resetField "pdfa" ("1b:2005" :: String)
```

Reply:
You could use `lookupMetaBool`

Reply:
That's what I thought at first, but the `pdfa` variable in this pull might contain a string now. (like `"1b:2005"`).

Reply:
When I try this, the `resetField` condition is never triggered, so that `-V pdfa` gives a literal `PDF/A-true` in the context document.  Note that `lookupMetaString` has type `String -> Meta -> String`, while `getField` is `FromJSON a => String -> Value -> Maybe a`.

Reply:
I believe `-V` didn't work with your version either? It should be `-M` see e.g. https://github.com/jgm/pandoc/blob/master/doc/customizing-pandoc.md#template-variables

Reply:
uh, yes... the `metadata` has the template variables (`-V`) already merged in, while the `meta :: Meta` only has the metadata variables (`-M`), see e.g. https://github.com/jgm/pandoc/blob/master/doc/customizing-pandoc.md#template-variables

I think it's okay if `-V pdfa` doesn't work, as long as `-M pdfa` and the following work:

```
---
pdfa: true
---

my markdown doc...
```

This is a bit confusing for the users, but we already have lots of places where `-V` doesn't work.

Reply:
Yes, both `-M pdfa` and `pdfa: true` in a YAML metadata block work.

Reply:
The following does it all, but it's really not very nice code...

```haskell
  let resetPDFA = resetField "pdfa" ("1b:2005"::String)
  let context =   ...

                $ (case getField "pdfa" metadata of
                        Just True             -> resetPDFA
                        _                     -> id)
                $ (case getField "pdfa" metadata of
                        Just ("true" :: String) -> resetPDFA
                        _                     -> id) metadata
```

I don't know... I'll let @jgm make a judgement call ;-)

Reply:
Apart from this, I think this pull is good to merge, as hmenke's concerns seem to have been addressed if I understand correctly. Thanks a lot!

Reply:
`-V pdfa=2a` will work regardless of what option is chosen here, and this should be the normal use case. The support for boolean `pdfa` (which necessitates the changes in the Context writer), is for backwards compatibility. Perhaps I should clarify the documentation on that.

Reply:
@jgm what do you think? apart from this minor issue, I think this is good to merge!

Reply:
I used the [first suggestion](https://github.com/jgm/pandoc/pull/5610#discussion_r300665428) made by @mb21 in the [last commit](https://github.com/jgm/pandoc/pull/5610/commits/e5c5bf12b11302a7d2cf5856a447a5fd861330fb). I have not found a solution that requires less code than [the later suggestion](https://github.com/jgm/pandoc/pull/5610#discussion_r300712630), and also works with `-V pdfa`, but I updated the manual to clarify on that.